infer/src/pulse/PulseFormula.ml

```diff
@@ -4093,10 +4252,11 @@ module Intervals = struct
         Sat (phi, new_eqs)
 end
 
-let and_atom atom formula =
+let and_atom atom formula add_term =
   let open SatUnsat.Import in
-  let* phi, new_eqs = Formula.Normalizer.and_atom atom (formula.phi, RevList.empty) in
-  let+ formula = Intervals.incorporate_new_eqs new_eqs {formula with phi} in
+  let* updphi, new_eqs = Formula.Normalizer.and_atom atom (formula.phi, RevList.empty) add_term in
```

Comment:
If we add the raw atom and term here ourselves then we don't need `Formula.Normalizer.and_atom` to do it.